nano/node/wallet.cpp

```diff
@@ -1455,23 +1455,6 @@ void nano::work_watcher::watching (nano::qualified_root const & root_a, std::sha
 
 								if (!ec)
 								{
-									{
-										auto hash (block_a->hash ());
-										nano::lock_guard<std::mutex> active_guard (watcher_l->node.active.mutex);
-										auto existing (watcher_l->node.active.roots.find (root_a));
-										if (existing != watcher_l->node.active.roots.end ())
-										{
-											auto election (existing->election);
-											if (election->status.winner->hash () == hash)
-											{
-												election->status.winner = block;
-											}
-											auto current (election->blocks.find (hash));
-											assert (current != election->blocks.end ());
-											current->second = block;
-										}
-									}
-									watcher_l->node.network.flood_block (block, false);
```

Comment:
Do we need flood_block forced here anyway since a different work value leads to the same hash?

Reply:
Block flooding is handled by election which is informed of the update via active.update_difficulty

Reply:
But in election.publish it will never enter the code that floods the block because the hash already existed from the block with the initial work.

But seems like this is a bug in the current implementation, updated blocks reaching other nodes are never republished, should we make it another PR? Perhaps checking if the work is updated [here](https://github.com/nanocurrency/nano-node/pull/2518/files#diff-2e3381130bf5f753e3cb73294268986eR222-R230) ?

Reply:
Alternatively could flood the block to _all_ PRs from the work watcher with no need to be republished.

Reply:
Placing the burden on the initial flood makes sense, in another PR.